### PR TITLE
[PW-2251] Expose the /paymentMethods API endpoints

### PR DESCRIPTION
**Pull request by msilvagarcia**
**Description**
This endpoint is needed to render the PM components in the checkout.

**Tested scenarios**
GET request to {{shopware-domain}}/sales-channel-api/v1/adyen/payment-methods with valid sw-access-key and sw-context-token returns PMs enabled for the merchant account configured.

**Fixed issue** PW-2251

### CODE REVIEW

File: src/Service/CurrencyService.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+     * @var EntityRepositoryInterface $currencyRepository
+     */
+    private $currencyRepository;
+
+    public function __construct(EntityRepositoryInterface $currencyRepository)
+    {
+        $this->currencyRepository = $currencyRepository;
+    }
+
+    public function getOrderCurrency(OrderEntity $order, Context $context): CurrencyEntity
```
Comment (msilvagarcia) [inline src/Service/CurrencyService.php:47]: Is this method used?

File: src/Service/PaymentMethodsService.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        $amount = $this->currency->sanitize($cart->getPrice()->getTotalPrice(), $currency);
+        if ($context->getCustomer()->getActiveBillingAddress()->getCountry()->getIso()) {
+            $countryCode = $context->getCustomer()->getActiveBillingAddress()->getCountry()->getIso();
+        } else {
+            $countryCode = $context->getCustomer()->getActiveShippingAddress()->getCountry()->getIso();
+        }
+        $shopperReference = $context->getCustomer()->getId();
+        $shopperLocale = $this->localeService->getLocaleFromLanguageId(
+            $context->getSalesChannel()->getLanguageId()
+        )->getCode();
```
Comment (msilvagarcia) [inline src/Service/PaymentMethodsService.php:102]: I remember we found a language code in `$context`. Is there something you forgot? Or did you find anything else about it?

File: src/Service/CurrencyService.php (hunk 1 above)
Comment (acampos1916) [reply src/Service/CurrencyService.php]: @msilvagarcia removed

File: src/Service/PaymentMethodsService.php (hunk 2 above)
Comment (acampos1916) [reply src/Service/PaymentMethodsService.php]: @msilvagarcia found a nicer way to get the locale through the sales channel search criteria

File: src/Service/CurrencyService.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ */
+
+namespace Adyen\Shopware\Service;
+
+use Adyen\Util\Currency;
+
+class CurrencyService extends Currency
```
Comment (cyattilakiss) [inline src/Service/CurrencyService.php:28]: Do you need this currency service? Isn't it easier just to use the service from the library? Or maybe you see any future purpose for it?

Comment (acampos1916) [reply src/Service/CurrencyService.php]: We can remove it for now 👍

**Pull request merged**